src/circuit/gadgets/macros.rs

```diff
@@ -209,11 +209,11 @@ macro_rules! or {
         )
     };
     ($cs:expr, $a:expr, $b:expr, $c:expr, $($x:expr),+) => {{
-        let or_tmp_cs_ =  &mut $cs.namespace(|| format!("or({})", stringify!(vec![$a, $b, $c, $($x),*])));
```

Comment:
I think the danger is pretty much code complexity, since things can grow pretty hairy with compound macros and their custom syntax. If abused inappropriately, debugging some macros can be hell

Reply:
Everyone should know about https://github.com/dtolnay/cargo-expand btw.